Author: Michal Srb <msrb@suse.com>
Subject: xfree86: Do not claim pci slots if fb slot is already claimed.
Patch-Mainline: To be upstreamed
References: bsc#1119431

The xf86PostProbe would terminate with fatal error if both fb and pci
slot were claimed at the same time, so there is no point in trying.
The opposite logic is already implemented - fb slots will not be claimed
if a pci slot was claimed.

This fixes issue with xf86-video-fbdev and xf86-video-vesa activating at
the same time (when /dev/fb0 is provided by the vesafb kernel driver).
---
 hw/xfree86/common/xf86pciBus.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/xfree86/common/xf86pciBus.c b/hw/xfree86/common/xf86pciBus.c
index 0718cdcb0..24396a63c 100644
--- a/hw/xfree86/common/xf86pciBus.c
+++ b/hw/xfree86/common/xf86pciBus.c
@@ -212,6 +212,9 @@ xf86ClaimPciSlot(struct pci_device *d, DriverPtr drvp,
     EntityPtr p = NULL;
     int num;
 
+    if (fbSlotClaimed)
+        return -1;
+
     if (xf86CheckPciSlot(d)) {
         num = xf86AllocateEntity();
         p = xf86Entities[num];
-- 
2.16.4

